tests/ui/type/type-name-basic.rs

```diff
@@ -62,28 +62,31 @@ pub fn main() {
 
     t!(Vec<Vec<u32>>, "alloc::vec::Vec<alloc::vec::Vec<u32>>");
     t!(Foo<usize>, "type_name_basic::Foo<usize>");
-    t!(Bar<'static>, "type_name_basic::Bar");
-    t!(Baz<'static, u32>, "type_name_basic::Baz<u32>");
+    t!(Bar<'static>, "type_name_basic::Bar<'_>");
+    t!(Baz<'static, u32>, "type_name_basic::Baz<'_, u32>");
 
-    // FIXME: lifetime omission means these all print badly.
-    t!(dyn TrL<'static>, "dyn type_name_basic::TrL<>");
-    t!(dyn TrLA<'static, A = u32>, "dyn type_name_basic::TrLA<, A = u32>");
+    t!(dyn TrL<'static>, "dyn type_name_basic::TrL<'_>");
+    t!(dyn TrLA<'static, A = u32>, "dyn type_name_basic::TrLA<'_, A = u32>");
     t!(
         dyn TrLT<'static, Cow<'static, ()>>,
-        "dyn type_name_basic::TrLT<, alloc::borrow::Cow<()>>"
+        "dyn type_name_basic::TrLT<'_, alloc::borrow::Cow<'_, ()>>"
     );
     t!(
         dyn TrLTA<'static, u32, A = Cow<'static, ()>>,
-        "dyn type_name_basic::TrLTA<, u32, A = alloc::borrow::Cow<()>>"
+        "dyn type_name_basic::TrLTA<'_, u32, A = alloc::borrow::Cow<'_, ()>>"
     );
 
     t!(fn(i32) -> i32, "fn(i32) -> i32");
-    t!(dyn for<'a> Fn(&'a u32), "dyn core::ops::function::Fn(&u32)");
+
+    // FIXME: these are sub-optimal, ideally the `for<...>` would be printed.
+    t!(dyn Fn(&'static u32), "dyn core::ops::function::Fn(&u32)");
+    t!(dyn for<'a> Fn(&'a u32), "dyn core::ops::function::Fn(&'0 u32)");
+    t!(dyn for<'a, 'b> Fn(&'a u32, &'b u32), "dyn core::ops::function::Fn(&'0 u32, &'1 u32)");
 
```

Comment:
```suggestion

    t!(fn(&'static u32), "fn(&u32)");
    t!(for<'a> fn(&'a u32), "fn(&'0 u32)");
    t!(for<'a, 'b> fn(&'a u32, &'b u32), "fn(&'0 u32, &'1 u32)");
    t!(for<'a> fn(for<'b> fn(&'a u32, &'b u32)), "fn(fn(&'0 u32, &'0 u32))");
   
```
printing only the `var` of the bound regions can result in duplicate bound regions. You would need to track some internal state in the printer which maps assigns each bound region a unique name.

I personally don't know whether printing bound regions is useful without printing their corresponding `for` and without differentiating between bound regions from different binders. I would prefer always using `'_` and putting a FIXME into `print_region` and then merging this PR, given that it's a very clear improvement otherwise

Reply:
Ok, I'll use `'_` for all printed regions.